src/Service/Search/SolrSuggest.php

```diff
@@ -0,0 +1,136 @@
+     */
+    public function suggest(SuggestQuery $query): SuggestResult
+    {
+        $client = $this->clientFactory->create($query->index);
+
+        $solrQuery = $this->buildSolrQuery($client, $query);
+        $solrResult = $client->select($solrQuery);
+        return $this->buildResult($solrResult);
+    }
+
+    private function buildSolrQuery(
+        Client $client,
+        SuggestQuery $query
+    ): SolrSelectQuery {
+        $solrQuery = $client->createSelect();
+        $solrQuery->addParam("spellcheck", "true");
+        $solrQuery->addParam("spellcheck.accuracy", "0.6");
+        $solrQuery->addParam("spellcheck.onlyMorePopular", "false");
+        $solrQuery->addParam("spellcheck.count", "15");
+        $solrQuery->addParam("spellcheck.maxCollations", "5");
+        $solrQuery->addParam("spellcheck.maxCollationTries", "15");
+        $solrQuery->addParam("spellcheck.collate", "true");
+        $solrQuery->addParam("spellcheck.collateExtendedResults", "true");
+        $solrQuery->addParam("spellcheck.extendedResults", "true");
+        $solrQuery->addParam("facet", "true");
+        $solrQuery->addParam("facet.sort", "count");
+        $solrQuery->addParam("facet.method", "enum");
+        $solrQuery->addParam(
+            "facet.prefix",
+            $query->text
+        );
+        $solrQuery->addParam("facet.limit", $query->limit);
+        $solrQuery->addParam("facet.field", self::INDEX_SUGGEST_FIELD);
+
+        $solrQuery->setOmitHeader(false);
+        $solrQuery->setStart(0);
+        $solrQuery->setRows(0);
+
+        // Filter
+        foreach ($query->filter as $filter) {
+            $solrQuery->createFilterQuery($filter->key)
+                ->setQuery($filter->getQuery())
+                ->setTags($filter->tags);
+        }
+
+        return $solrQuery;
+    }
+
+    private function buildResult(
+        SolrSelectResult $solrResult
+    ): SuggestResult {
+        $suggestions = $this->parseSuggestion(
+            $solrResult->getResponse()->getBody()
+        );
+        return new SuggestResult(
+            $suggestions,
+            $solrResult->getQueryTime() ?? 0
+        );
+    }
+
+    /**
+     * @throws UnexpectedResultException
+     * @return Suggestion[]
+     */
+    private function parseSuggestion(
+        string $responseBody
+    ): array {
+        try {
+            /** @var SolariumResponse $json */
+            $json = json_decode(
+                $responseBody,
+                true,
+                5,
+                JSON_THROW_ON_ERROR
+            );
+            $facets =
+                $json['facet_counts']['facet_fields'][self::INDEX_SUGGEST_FIELD]
+                ?? [];
+
+            $len = count($facets);
+
+            $suggestions = [];
+            for ($i = 0; $i < $len; $i += 2) {
+                $term = $facets[$i];
+                $hits = (int)$facets[$i + 1];
+                $suggestions[] = new Suggestion($term, $hits);
+            }
```

Comment:
if `$len` is uneven we run into an error here - could prevent it if we do one of these:
```php
for ($i = 0; $i + 1 < $len; $i += 2) {
```
```php
for ($i = 0; $i < $len - 1; $i += 2) {
```
```php
$len = count($facets) - 1;
```